src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -57,6 +57,18 @@ function getUnmaskedContext(workInProgress : Fiber) : Object {
 }
 exports.getUnmaskedContext = getUnmaskedContext;
 
+function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
+  if (isContextConsumer(workInProgress)) {
```

Comment:
We already doing an equivalent check in `getMaskedContext` which is called just before `cacheContext`.
Can we avoid doing it twice?